### ISSUE TITLE
ResolvedDateTimeFormatOptions Timezone is possibly undefined

### ISSUE THREAD

**Issue by aar9nk** [microsoft/TypeScript#55869]
### ⚙ Compilation target

ES2015

### ⚙ Library

 4.9.5

### Missing / Incorrect Definition

`ResolvedDateTimeFormatOptions`'s `Timezone` is possibly undefined when using various browsers. This can happen when the user has not set a timezone in their browser ([such as chrome](https://support.google.com/chrome/thread/231926653/timezone-return-undefined?hl=en)). 

### Sample Code

```TypeScript
`const isInEu = Intl.DateTimeFormat().resolvedOptions().timeZone.startsWith("Europe") ? true : false;` 
```
will cause an error for a user who has not specified their browser timezone and cause a crash. This could be handled with an undefined however typescript does not currently give me the option

### Documentation Link

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/resolvedOptions#:~:text=timeZone,time%20zone%20instead.

**Comment by Josh-Cena** [comment]
This has been changed and is now never `undefined`. See: https://tc39.es/ecma402/#sec-intl.datetimeformat.prototype.resolvedoptions

> In this version of the API, the "timeZone" property will be the identifier of the [host environment](https://tc39.es/ecma262/#host-environment)'s time zone if no "timeZone" property was provided in the options object provided to the Intl.DateTimeFormat [constructor](https://tc39.es/ecma262/#constructor). The first edition left the "timeZone" property undefined in this case.

MDN hasn't been updated, but I'm working on it already.

**Comment by aar9nk** [comment]
I've created a [subsequent PR](https://github.com/microsoft/TypeScript/pull/55870) 55870 if anyone get's a chance to review it :)

**Comment by Josh-Cena** [comment]
Note that the current spec already says it should always return a string: https://tc39.es/ecma402/#sec-intl.datetimeformat.prototype.resolvedoptions

> In this version of the API, the "timeZone" property will be the identifier of the host environment's time zone if no "timeZone" property was provided in the options object provided to the Intl.DateTimeFormat constructor. The first edition left the "timeZone" property undefined in this case.

The way the spec describes this part, it appears that `timeZone` can never be undefined (see also https://tc39.es/ecma402/#sec-properties-of-intl-datetimeformat-instances, which says [[TimeZone]] is always a string). I wonder if Chrome is being compliant here, but I don't know how to unset my system time zone.

I'm halfway through updating MDN's `resolvedOptions()` pages, and the paragraph you quoted will surely be updated, so I want to be sure that it matches both the spec and reality.

**Comment by aar9nk** [comment]
> Note that the current spec already says it should always return a string: https://tc39.es/ecma402/#sec-intl.datetimeformat.prototype.resolvedoptions
> 
> > In this version of the API, the "timeZone" property will be the identifier of the host environment's time zone if no "timeZone" property was provided in the options object provided to the Intl.DateTimeFormat constructor. The first edition left the "timeZone" property undefined in this case.
> 
> The way the spec describes this part, it appears that `timeZone` can never be undefined (see also https://tc39.es/ecma402/#sec-properties-of-intl-datetimeformat-instances, which says [[TimeZone]] is always a string). I wonder if Chrome is being compliant here, but I don't know how to unset my system time zone.
> 
> I'm halfway through updating MDN's `resolvedOptions()` pages, and the paragraph you quoted will surely be updated, so I want to be sure that it matches both the spec and reality.

Yeah after some testing it appears like you are correct. I'm happy to close this and open an issue with chrome! I appreciate the expertise

**Comment by duncannz** [comment]
@aar9nk I am experiencing this issue with Chrome too (on macOS Sonoma), and was wondering if you ended up opening a Chrome bug for it?

**Comment by spherebread** [comment]
@duncannz seems like it's fixed in Chrome 118.0.5993.88

**Comment by duncannz** [comment]
Can confirm the same, thank you!
